src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1260,6 +1260,65 @@ internal virtual FieldSymbol FixedElementField
         /// <returns>True if this is an interface type.</returns>
         internal abstract bool IsInterface { get; }
 
+        /// <summary>
+        /// Verify if the given type can be used to back a tuple type and return cardinality of that tuple type in <paramref name="tupleCardinality"/>. 
+        /// </summary>
+        /// <param name="tupleCardinality">If method returns true, contains cardinality of the compatible tuple type.</param>
+        /// <returns></returns>
+        public override bool IsTupleCompatible(out int tupleCardinality)
```

Comment:
Small thing, but since this is a public API: tuple "arity", "cardinality", "length". We'll want to pick one and use consistently.